### ISSUE TITLE
Without mysql driver, MySQLContainer keeps waiting instead fail fast

### ISSUE THREAD

**Issue by ryukato** [testcontainers/testcontainers-java#1797]
when `mysql-connector-java` depedenecy is not set, then it looks like test-container keep waiting for logger time. (it look like hang :( ). I think it throws a sort of an exception and stop.
Could you guyes take a look at the issue? I put some more details below.

Details to reproduce issue

Language: Kotlin
Test container version: 1.11.3
TestContainer Module: MySQLContainer

Pre-condition
* Not adding no mysql-connector depedenecy

Sample Codes
```
@RunWith(SpringRunner::class)
@SpringBootTest()
class TestWithMysqlContainerModule {

    companion object {
        const val TEST_DATABASE_NAME = "test"

        val mysqlContainer = KtMySQLContainer()
                .withExposedPorts(3306)
                .withDatabaseName(TEST_DATABASE_NAME)
                .withUsername("test_user")
                .withPassword("test_password")
                .withEnv("MYSQL_ROOT_PASSWORD", "test_password")

        fun start() {
            if (!mysqlContainer.isRunning) {
                mysqlContainer.start()
            }
        }

        fun stop() {
            if (!mysqlContainer.isRunning) {
                mysqlContainer.start()
            }
        }

        fun getMySQLConnectionUrl(): String {
            return mysqlContainer.jdbcUrl
        }
    }

    @Before
    fun setUp() {
        start()
    }

    @After
    fun tearDown() {
        stop()
    }


    @Test
    fun printMySQLConnectionUrl() {
        println(getMySQLConnectionUrl())
    }


    class KtMySQLContainer(imageName: String) : MySQLContainer<KtMySQLContainer>(imageName) {
        constructor() : this("mysql:5.7.27")
    }
}
```

**Comment by bsideup** [comment]
Hi @ryukato,

Could you please share your Testcontainers version?

**Comment by ryukato** [comment]
I updated issue description with the version.

**Comment by bsideup** [comment]
@ryukato please try the latest version, @rnorth implemented better handling of drivers

**Comment by ryukato** [comment]
Ok thanks I will check with the latest one.

**Comment by rnorth** [comment]
Yep, #1434, (released in 1.12.0) should improve this!

**Comment by ryukato** [comment]
Good! it is working beautifully. :) I will close this, Thanks again. 👍
